File: starterpack/microchallenge/python/MyBot.py
import argparse

from flask import (
    Flask,
    request
)

app = Flask(__name__)

@app.route('/microchallenge')
def microchallenge():
    print("\n\n\n-------------------------- REQUEST LOGS STARTING HERE --------------------------")
    print("You can log stuff and download the logs from the UI in the replay section.")
    print("Here is the current problem:")
    print(request.args.get('problem')); # problem is in json format
    print("---------------------------------------------------------------------------------")
    
    return str(18)


if __name__ == '__main__':
    parser = argparse.ArgumentParser(description='Microchallenge Starter Pack')
    parser.add_argument('-p', metavar='p', type=int)

    args = parser.parse_args()
    app.run(port=args.p, host="0.0.0.0")

File: starterpack/microchallenge/typescript/MyBot.ts
import * as express from 'express';
import {Request, Response} from 'express';
import * as yargs from 'yargs';

const app = express();
var argv = yargs
    .number('p')
    .number('port')
    .argv
const PORT = argv.port || argv.p || 8010;

app.get('/microchallenge', (req:Request, res:Response) => {
    console.log("\n\n\n-------------------------- REQUEST LOGS STARTING HERE --------------------------");
    console.log("You can log stuff and download the logs from the UI in the replay section.");
    console.log("Here is the current problem:");
    console.log(req.query.problem); // problem is in json format
    console.log("---------------------------------------------------------------------------------");

    res.send((18).toString());
});

app.listen(PORT, () => {
    console.log(`Example app listening on port ${PORT}!`)
});
